Elements/src/Spatial/CellComplex.cs

```diff
@@ -0,0 +1,1285 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Elements;
+using Elements.Geometry;
+using Newtonsoft.Json;
+
+namespace Elements.Spatial
+{
+    #region subclasses
+
+    /// <summary>
+    /// A unique vertex in a cell complex
+    /// </summary>
+    public class Vertex
```

Comment:
Per our conversation of today, to mark these `internal` we'd need to also mark the dictionaries on `CellComplex` for lookup as `internal`, which may not be what you want. The question is how we expect a user to interact with the `CellComplex`. If we go to a model where we can traverse from anything to anything, then the dictionaries on `CellComplex` are only required internally because a user can do:
```csharp
// Find a vertical edge from a given vertex.
var verticalEdge = myVertex.Edges.First(e => e.Direction.IsParallelTo(Vector3.ZAxis));

// Find the vertex at the other end of this edge.
var v = verticalEdge.Start == myVertex ? verticalEdge.End : verticalEdge.Start;

// Run this logic again on the found vertex to continue traversing.
...
```